sympy/solvers/tests/test_polysys.py

```diff
@@ -197,11 +197,11 @@ def test_factor_system():
     assert result == expected
 
     result = factor_system([x ** 2 + 1, y ** 2 + 1], gaussian=True)
-    expcted = ([[x + I, y - I], [x + I, y + I], [x - I, y - I], [x - I, y + I]], True)
+    expcted = ([[y - I, x - I], [y - I, x + I], [y + I, x - I], [y + I, x + I]], True)
```

Comment:
Why have all of the orderings changed?

Reply:
It happens when we refactor `_has_common_variables` probably because that's just how `_cnf2dnf` works. 

We fixed how polynomials are checked for shared variables. What I think is, for example we had `x**2` and `y`, they were considered to share variables (because they both had `x,y` as gens), so they were processed together. 

Although now we correctly identify they're independent (no variables with non-zero degrees in common), so they're processed separately through `cartes`.

Correct me if I'm thinking in the wrong direction here.

Reply:
I think we should make it so that the cartes and non-cartes code paths both produce the subsystems in the same order. Something like this:
```diff
diff --git a/sympy/solvers/polysys.py b/sympy/solvers/polysys.py
index f25af9b8f6..c9e3388922 100644
--- a/sympy/solvers/polysys.py
+++ b/sympy/solvers/polysys.py
@@ -812,7 +812,7 @@ def _cnf2dnf(eqs):
     # remove equations that are independent of all others
     # and generate the Cartesian product for them
     eqs = [list(i) for i in eqs]
-    indep = []
+    indep = set()
 
     for i in range(len(eqs)):
         is_independent = True
@@ -829,16 +829,23 @@ def _cnf2dnf(eqs):
             if not is_independent:
                 break
         if is_independent:
-            indep.append(i)
+            indep.add(i)
 
     if not indep:
         return list(_dnf(eqs))
     else:
+        indep_eqs = []
+        dep_eqs = []
+        for i, e in enumerate(eqs):
+            if i in indep:
+                indep_eqs.append(e)
+            else:
+                dep_eqs.append(e)
+
         result = []
-        indep_eqs = [eqs.pop(i) for i in indep[::-1]]
         for t in cartes(*indep_eqs):
             t_list = list(t)
-            for d in _dnf(eqs):
+            for d in _dnf(dep_eqs):
                 result.append(t_list + d)
         return result
```